### PR TITLE
Build each subproject with travis

### PR DESCRIPTION
**Pull request by dckc**
This is another take at #54 (and #52), after re-basing against dev to get the right storage bits.

Noteably, we use a **separate parallel matrix travis build for each subproject** (8b51740). See, for example [build 16](https://travis-ci.org/rchain/rchain/builds/306142977), with 16.1 thru 16.5.

I looked into `$TRAVIS_COMMIT_RANGE` but I'm inclined to leave that sort of optimization to later.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+dist: trusty
+install:
+        - sudo apt update
+
+        # Packages for Rosette-VM building
+        - sudo apt install g++ g++-multilib
```
Comment (DoSOfRedRiver) [inline .travis.yml]: I think that we should move the bottom level dependencies to concrete subprojects builds, so we keep .travis.yml as clear as possible and leave here only Travis related settings.
So if it possible I suggest moving this "apt-get install" stuff to concrete build.sh files.

**Comment by DoSOfRedRiver** [comment]
Btw did you try the $TRAVIS_COMMIT_RANGE or we decided to figure it out later?

**Comment by dckc** [comment]
Now that I have read the docs on `$TRAVIS_COMMIT_RANGE`, I'm inclined to figure it out later. The docs say, for example: 
> Note that this is empty for builds triggered by the initial commit of a new branch.

Aren't most PRs on new branches? If I make a `feature-XYZ` branch and then make a PR for that branch against `dev`, are they talking about `feature-XYZ` or `dev`?

Now that build 9 completed, I'm inclined to postpone optimizations like this 'till after this PR lands.

File: .travis.yml (hunk 1 above)
Comment (dckc) [reply .travis.yml]: I'm not sure what you mean. There is no `rosette/build.sh`.

Comment (dckc) [reply .travis.yml]: I think I see now... I'm trying matrix builds, too.

**Comment by dckc** [comment]
Thanks, @DoSOfRedRiver. Who has the ball now? I gather you guys normally merge your own PRs, but I don't have access to do that.

**Comment by DoSOfRedRiver** [comment]
@dckc I can merge it if you wish.

**Comment by MParlikar** [comment]
Let's get this PR Merged please.

**Comment by dckc** [comment]
@DoSOfRedRiver did you do a squash merge? It makes the history really confusing... eebbdd8 attributes a whole bunch of stuff to me that is actually from Kent's stuff that I rebased.

Do you guys want stuff squashed? I asked in #40 but didn't hear anything. If you want stuff squashed, I'll try to do that so that you don't have to do a squash merge.

**Pull request merged**